### PR TITLE
Assign lastEventId only if event was queued for submission

### LINKED ISSUE #1414: Assign lastEventId only if we were able to send the event
right now we assign the lastEventId even if the SDK is disabled

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Assign lastEventId only if event has been sent successfully

## :bulb: Motivation and Context

Fixes #1414

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * Fix: Clock drift issue when calling DateUtils#getDateTimeWithMillisPrecision (#1557)
 * Feat: Set mechanism type in SentryExceptionResolver (#1556)
 * Ref: Prefer snake case for HTTP integration data keys (#1559)
+* Feat: Assign lastEventId only if event has been sent successfully (#1565)
```
Comment (bruno-garcia) [inline CHANGELOG.md:14]:
```suggestion
* Feat: Assign lastEventId only if event was queued for submission (#1565)
```

**Review by bruno-garcia — APPROVED** [review]
Better call it 'queue for submission' or similar since it can still be dropped due to date limiting, beforeSend, eventProcessor or some error

File: CHANGELOG.md (hunk 1 above)
Comment (maciejwalkowiak) [reply CHANGELOG.md:14]: Agreed!

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 * Fix: Clock drift issue when calling DateUtils#getDateTimeWithMillisPrecision (#1557)
 * Feat: Set mechanism type in SentryExceptionResolver (#1556)
 * Ref: Prefer snake case for HTTP integration data keys (#1559)
+* Feat: Assign lastEventId only if event was queued for submission (#1565)
```
Comment (marandaneto) [inline CHANGELOG.md:14]:
```suggestion
* Fix: Assign lastEventId only if event was queued for submission (#1565)
```

sounds more like a Fix

**Review by marandaneto — APPROVED** [review]

**Pull request merged**